xds/src/main/java/io/grpc/xds/XdsClient.java

```diff
@@ -55,39 +56,39 @@
     // The list virtual hosts that make up the route table.
     @Nullable
     final List<VirtualHost> virtualHosts;
-    // Listener contains the HttpFault filter.
-    final boolean hasFaultInjection;
-    @Nullable // Can be null even if hasFaultInjection is true.
-    final HttpFault httpFault;
+    // Filter instance names. Null if HttpFilter support is not enabled.
+    @Nullable final List<String> filterChain;
```

Comment:
Why not have `Filter` contains `FilterConfig`? Then here you would only need a single list of `Filter`s. And in the resolver, you do not need to look up the global registry again.

Reply:
Did you mean have `FilterConfig` contain `Filter`?

Reply:
See my comment for Filter below. I mean something like

```
interface Filter {
  // Returns the canonical name of the filter.
  String getName();

  FilterConfig getConfig();
  
  interface Parser {
    // Returns the list of typeUrls for filter configs that are acceptable by this parser.
    String[] getTypeUrls();

   FilterConfig parseConfig(Any config);
  }

  ...
}
```

Generics can probably make this better.

Reply:
No, the filter config override is finalized in XdsNameResolver's `selectConfig()`. Way too late to instantiate such an object.

Reply:
At the minimum, I think you could make the `FilterConfig` have a property for the name of filter it is associated with. Then you can eliminate this List + Map representation.

Reply:
As a ref, C-core's representation looks simpler to manage: https://github.com/grpc/grpc/blob/0937cb1249129a84159c639c59f49a451157edf7/src/core/ext/xds/xds_api.h#L231-L242

Reply:
These two approaches are just preferences. It's not that one of them is significantly simpler than other. For the FilterConfig with name approach, it's inconvenient to query the name and override config.